opensearch-operator/controllers/cluster_test.go

```diff
@@ -113,9 +113,19 @@ var _ = Describe("Cluster Reconciler", func() {
 				}))
 
 			// check if the ServiceMonitor is using the interval from the CRD declaration
-
 			Expect(sm.Spec.Endpoints[0].Interval).Should(BeEquivalentTo(OpensearchCluster.Spec.General.Monitoring.ScrapeInterval))
 
+			// check if the ServiceMonitor adheres to the CRD definition of tlsConfig and doesn't define it if not defined in the CRD
+			if OpensearchCluster.Spec.General.Monitoring.TLSConfig != nil {
+				// check if the ServiceMonitor is using the tlsConfig.insecureSkipVerify from the CRD declaration
+				Expect(sm.Spec.Endpoints[0].TLSConfig.InsecureSkipVerify).Should(BeEquivalentTo(OpensearchCluster.Spec.General.Monitoring.TLSConfig.InsecureSkipVerify))
+
+				// check if the ServiceMonitor is using the tlsConfig.serverName from the CRD declaration
+				Expect(sm.Spec.Endpoints[0].TLSConfig.ServerName).Should(BeEquivalentTo(OpensearchCluster.Spec.General.Monitoring.TLSConfig.ServerName))
+			} else {
+				// check if tlsConfig is not defined in the CRD declaration the ServiceMonitor not deploy that part of the config
+				Expect(sm.Spec.Endpoints[0].TLSConfig).To(BeNil())
+			}
```

Comment:
As a general rule tests should not do such "if expect else expect" checks. The test should be setup in a specific way (in your case it's probably best to have two tests, one for tlsconfig and one without) and then directly check for the expected outcome.

Reply:
@swoehrl-mw Hi, hopefully I have interpretted your comment correctly, please let me know

Reply:
@rhysxevans There should be no if at all. Just add a TLSConfig to the cluster definition in `suite_test_helpers.go` (`ComposeOpensearchCrd`) and then move your expects out of the ifs and remove the ifs. Because at the moment your tests are not executed because in the used cluster spec no tlsconfig is defined.

Reply:
@swoehrl-mw Hi, hopefully I am on the correct path now? Please note I am not sure how to test for `nil` in this setup (see comments on last push)